include/vikunja/mem/iterator/ConstantIterator.hpp

```diff
@@ -0,0 +1,267 @@
+                }
+
+                /**
+                 * @brief Postfix decrement operator
+                 * @note Use prefix decrement operator instead if possible to avoid copies
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator--()
+                {
+                    ConstantIterator cpy = *this;
+                    --index;
+                    return cpy;
+                }
+
+                /**
+                 * @brief Prefix decrement operator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator--(int)
+                {
+                    --index;
+                    return *this;
+                }
+
+                /**
+                 * @brief Add an index to this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator+(const IdxType idx) const
+                {
+                    return ConstantIterator(v, index + idx);
+                }
+
+                /**
+                 * @brief Add a second constant iterator of the same value to this one
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator+(const ConstantIterator& other) const
+                {
+                    assert(v == other.v && "Can't add constant iterators of different values!");
+                    return ConstantIterator(v, index + other.index);
+                }
+
+                /**
+                 * @brief Subtract an index from this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator-(const IdxType idx) const
+                {
+                    return ConstantIterator(v, index - idx);
+                }
+
+                /**
+                 * @brief Subtract a second constant iterator of the same value from this one
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator-(const ConstantIterator& other) const
+                {
+                    assert(v == other.v && "Can't subtract constant iterators of different values!");
+                    return ConstantIterator(v, index - other.index);
+                }
+
+                /**
+                 * @brief Add an index to this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator& operator+=(const IdxType idx)
+                {
+                    index += idx;
+                    return *this;
+                }
+
+                /**
+                 * @brief Subtract an index from this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator& operator-=(const IdxType idx)
+                {
+                    index -= idx;
+                    return *this;
+                }
+
+#pragma endregion arithmeticoperators
+
+#pragma region comparisonoperators
+
+// if spaceship operator is available is being used we can use spaceship operator magic
```

Comment:
```suggestion
// if spaceship operator is available we can use spaceship operator magic
```
Or also just drop it, because it's clear from the code what you do.